src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { TaskStatus } from './types/types';
+
+type FilterOptions = {
+  query: string;
+  taskStatusFilter: TaskStatus;
+};
+
+const getFilteredTodos = (todos: Todo[], options: FilterOptions) => {
+  const { query, taskStatusFilter } = options;
+  let filteredTodos = [...todos];
+
+  const normalizeStr = (str: string): string => str.trim().toLowerCase();
+
+  if (query) {
+    filteredTodos = filteredTodos.filter(todo =>
+      normalizeStr(todo.title).includes(normalizeStr(query)),
+    );
+  }
+
+  switch (taskStatusFilter) {
+    case TaskStatus.Active:
+      return filteredTodos.filter(todo => !todo.completed);
+    case TaskStatus.Completed:
+      return filteredTodos.filter(todo => todo.completed);
+    default:
+      return filteredTodos;
+  }
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoaded, setIsLoaded] = useState(false);
```

Comment:
```suggestion
  const [isLoading, setIsLoading] = useState(false);
```
To show the loader you should set this state to true when you start loading and then remove it after you get a response from the server in finally block